### PR TITLE
Added smartWarning, which does deduplication intelligently.

### PR DESCRIPTION
**Pull request by jimfb**
Added smartWarning, which does deduplication intelligently.

### CODE REVIEW

**Comment by jsfb** [comment]
cc @sebmarkbage

File: src/vendor/core/smartWarning.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+var emptyFunction = require('emptyFunction');
+
+/**
+ * Similar to warning except warnings are intelligently deduplicated.
+ */
+
+var smartWarning = emptyFunction;
+
+var cache = {};
+var hasOwnProperty = {}.hasOwnProperty;
```
Comment (SanderSpies) [inline src/vendor/core/smartWarning.js]: Not used anywhere.

File: src/vendor/core/smartWarning.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  cache[cacheKey] = list;
+  return true;
+}
+
+if (__DEV__) {
+  smartWarning = function(condition, format, ...args) {
+    var argIndex = 0;
+    var message = 'Warning: ' + format.replace(/%s/g, () => args[argIndex++]);
+    var cacheKey;
+    if (args.length-1 === argIndex) {
```
Comment (SanderSpies) [inline src/vendor/core/smartWarning.js]: Spacing before and after `-`

**Comment by zpao** [comment]
We went in another direction with warnings so just going to close this out.

**Pull request closed without merging**